Android: Update dependencies

m1ga opened the pull request:
**Update events:**
* `subscribe` & `unsubscribe` event with `(un)subscribed:true/false` 
* `tokenRemoved*

**New methods**
* deleteToken()
* getToken()
* clearLastData() - empty the stored lastData values

**Update dependencies:**
* com.google.firebase:firebase-messaging:21.0.0

Android library changelog:

* 21.0.0
* 20.3.0:  deleteToken, getToken in FCM (https://firebase.google.com/support/release-notes/android#messaging_v20-3-0)
* 20.2.4:	internal
* 20.2.3:	Fixed an issue that caused an app to crash when a user tapped on a received notification.
* 20.2.2:	internal (don't use)
* 20.2.1:	internal
* 20.2.0:	internal
* 20.1.7:	internal

hansemannn commented on android/src/firebase/cloudmessaging/CloudMessagingModule.java:
```diff
@@ -126,14 +126,36 @@ public void onComplete(@NonNull Task<InstanceIdResult> task)
+			public void onComplete(@NonNull Task<Void> task) {
+				KrollDict data = new KrollDict();
+				if (!task.isSuccessful()) {
+					data.put("unsubscribed", false);
```
Wouldn't a `success` flag be more simple here?

m1ga replied:
sounds better! Then it would be the same for un/subscribe

m1ga commented:
@hansemannn: done, please test it, then it is ready to be released

added the event as "Android only" at the moment. So you can add it to the iOS module later on

hansemannn commented:
Thanks! Have you validated that push messages (both data and clssic messages) as well as the intent handling / cold boot receival still works? Maybe someone can help here? I wont make it until next week

m1ga commented:
My test case sends out 4 notifications:
* normal notification
* normal notification + data
* normal notification to channel
* data notification (force foreground)

in foreground, background and closed app. 
And of course opening the app using the intent.

But it would be good if someone else does a quick test. Since it is nothing important this can wait, too. The current version still works!

hansemannn commented:
Yep, especially because it works fine without any issues, we should use the time to have it tested by a broader audience. Maybe release a beta version of it (and the related modules)?

jgiunta commented:
Hello guys. When will merge this PR ? I test it and works fine. Thanks

m1ga commented:
@jgiunta Thank you for the feedback! We will merge it soon

m1ga commented:
some small update:
* removed the BoM part. 

I've spoked to Joshau and the BoM part is mainly for the app gradle, not for the module gradle. So the user needs to add it. But there aren't that many benefits since the use mostly won't update that version. 
So to make it easier for the user we should just use the BoM list to get the matching versions and put them into the gradle file.

* added getToken() and deleteToken() methods. New in FCM 20.3.0 (https://firebase.google.com/support/release-notes/android#messaging_v20-3-0); refactored `registerForPushNotifications` to use `getToken` now.

* added `tokenRemoved` event
* added `clearLastData()` to empty the lastData by hand

[firebase.cloudmessaging-android-3.0.2.zip](https://github.com/hansemannn/titanium-firebase-cloud-messaging/files/5433418/firebase.cloudmessaging-android-3.0.2.zip)

m1ga commented:
* updated to a newer version
* tested it with 9.2.2 and 9.3.0 + ti.playservices 17.5.0 -> working fine

[firebase.cloudmessaging-android-3.0.2.zip](https://github.com/hansemannn/titanium-firebase-cloud-messaging/files/5541228/firebase.cloudmessaging-android-3.0.2.zip)

hansemannn commented:
@m1ga Can you update to 21.0.1? I'm fine with merging then! :)
